lib/hl7/mirth_xml_parser.rb

```diff
@@ -0,0 +1,121 @@
+require "nokogiri"
+
+module HL7
+  class MirthXmlParser
+    def initialize(xml, segment_delimiter: "\r")
+      @doc = Nokogiri::XML(xml)
+      @segment_delimiter = segment_delimiter
+    end
+
+    def to_hl7
+      segments = @doc.root.element_children
+      msh_segment, delimiters = parse_msh(segments.first)
```

Comment:
`shift` mutates, which is why I tend to avoid it. But I'd be okay with:
```ruby
first, *rest = @doc.root.element_children
msh_segment, delimiters = parse_msh(first)
other_segments = rest.map...
```

Reply:
yeah I had a feeling you'd have an issue with it because of the mutation :P

Reply:
I'm okay with what you already have, or what you have suggested.